app/controllers/api_controller.rb

```diff
@@ -2,8 +2,8 @@
 
 class ApiController < ApplicationController
   before_action :doorkeeper_authorize!
-  before_action :restrict_content_type
-  after_action :set_content_type
+  before_action :restrict_content_type, unless: -> { @force_content_type }
+  after_action :set_content_type, unless: -> { @force_content_type }
```

Comment:
Hi @ngw, just taking the ability to set response content-type explicitly, the flag used here feels a bit convoluted, and it's easy to see someone copy/pasting the `content_type` and then wondering why it's not being honoured. 

Could we just make this a `before_action` (without the `unless`? That way it's easy to override it explicitly with `content_type: ApiController::CONTENT_TYPE` (as you have) without needing to also say you don't want to set it for anything in that controller.

Reply:
Given the update to the description, I think the change I've suggested here would resolve your issue entirely (without changing `restrict_content_type`).

`restrict_content_type` prevents requests from arriving at the API with an incorrect `Content-Type`. If I'm understanding correctly, that doesn't sound like an issue here as you're expecting the correct `Content-Type`, but returning a different one.